java/core/src/main/java/co/worklytics/psoxy/storage/impl/ColumnarBulkDataSanitizerImpl.java

```diff
@@ -162,94 +184,146 @@ public void sanitize(@NonNull Reader reader,
                 "\"" + String.join("\",\"", headersCI) + "\""));
         }
 
-
-        List<String> columnNamesForOutputFile = Streams.concat(
-            applyReplacements(headers, columnsToRename).stream(),
-            columnsToDuplicate.values().stream())
-            .collect(Collectors.toList());
-
-
         TriFunction<String, String, Pseudonymizer, String> pseudonymizationFunction = buildPseudonymizationFunction(rules);
 
-        BiFunction<String, String, String> applyPseudonymizationIfAny = (outputColumnName, value) -> {
-            if (columnsToPseudonymize.contains(outputColumnName.toLowerCase())
-                || columnsToPseudonymizeIfPresent.contains(outputColumnName.toLowerCase())) {
-                return pseudonymizationFunction.apply(value, outputColumnName, pseudonymizer);
-            }
-            return value;
-        };
-
         Map<String, Pseudonymizer> pseudonymizers = new HashMap<>();
-        BiFunction<String, String, String> applyTransformIfAny = (originalColumnName, value) -> {
-          if (columnsToTransform.containsKey(originalColumnName.toLowerCase())) {
-              FieldTransformPipeline pipeline = columnsToTransform.get(originalColumnName.toLowerCase());
-
-              for ( FieldTransform transform : pipeline.getTransforms()) {
-                  if (value != null) {
-                      if (transform instanceof FieldTransform.Filter) {
-                          Pattern pattern = Pattern.compile(((FieldTransform.Filter) transform).getFilter());
-                          Matcher matcher = pattern.matcher(value);
-                          if (matcher.matches()) {
-                              if (matcher.groupCount() > 0) {
-                                  value = matcher.group(1);
-                              }
-                          } else {
-                              value = null;
+        BiFunction<String, FieldTransformPipeline, String> applyTransform = (originalValue, pipeline) -> {
+          String value = originalValue;
+          for ( FieldTransform transform : pipeline.getTransforms()) {
+              if (value != null) {
+                  if (transform instanceof FieldTransform.Filter) {
+                      Pattern pattern = Pattern.compile(((FieldTransform.Filter) transform).getFilter());
```

Comment:
Can we have this stored in a map, just to avoiding compile same patterns again and again in the same execution ?

Reply:
Same in L221